multipledispatch/tests/test_dispatcher.py

```diff
@@ -258,3 +261,139 @@ def _(a):
         raise MDNotImplementedError()
 
     assert raises(NotImplementedError, lambda: f(1.0))
+
+
+def test_vararg_not_last_element_of_signature():
+    f = Dispatcher('f')
+    assert raises(TypeError, lambda: f.register([float], str)(lambda: None))
+
+
+def test_vararg_has_multiple_elements():
+    f = Dispatcher('f')
+    assert raises(TypeError, lambda: f.register([float, str])(lambda: None))
+
+
+def test_vararg_dispatch_simple():
```

Comment:
should we add a test where we have a vararg registered and pass incompatible arguments to make sure it still gives us a notimplemented?

Reply:
For sure.

Reply:
Added